libopae/plugins/xfpga/metrics/metrics.c

```diff
@@ -310,23 +344,33 @@ fpga_result __FPGA_API__ xfpga_fpgaGetMetricsByName(fpga_handle handle,
 	if (objtype == FPGA_ACCELERATOR) {
 		// get AFU metrics
 		for (i = 0; i < num_metric_names; i++) {
-
 			result = parse_metric_num_name(metrics_names[i],
 							&(_handle->fpga_enum_metric_vector),
 							&metric_num);
 			if (result != FPGA_OK) {
-				FPGA_ERR("Invalid Input Metrics string");
+				FPGA_ERR("Invalid input metrics string= %s", metrics_names[i]);
+				metrics[i].metric_num = 0xFFFF;
 				continue;
 			}
 
 			result = get_afu_metric_value(handle, &(_handle->fpga_enum_metric_vector),
 				metric_num,
 				&metrics[i]);
 			if (result != FPGA_OK) {
-				FPGA_ERR("Failed to get metric value");
+				FPGA_ERR("Failed to get metric value  for metric = %s", metrics_names[i]);
+				metrics[i].metric_num = 0xFFFF;
```

Comment:
Will this result in an array index out of bounds somewhere later?

Reply:
user sends metrics search string as input , if Metric API couldn't found any metric with name.
struct fpga_metric {
	uint64_t metric_num;    // Metric index num
	metric_value value;     // Metric value
};

OPAE API fills struct fpga_metric with invalid index 0xFFF for not found metric by name.

Reply:
Can we use a #define name instead?

Reply:
fixed